packages/kit/src/utils/routing.js

```diff
@@ -63,11 +63,10 @@ export function parse_route_id(id) {
 										}
 
 										const match = param_pattern.exec(content);
-										if (!match) {
-											throw new Error(
-												`Invalid param: ${content}. Params and matcher names can only have underscores and alphanumeric characters.`
-											);
-										}
+										// Skip invalid params.
+										// Params and matcher names can only have underscores and alphanumeric characters.
+										// We'll never hit this because mnifest generation would have failed.
+										if (!match) return;
```

Comment:
if we never hit it, why do we have this line at all? surely a type assertion would be sufficient?

we should probably be separating these changes into different PRs — this isn't related to error messages (which i think we should solve properly rather than with piecemeal changes), and the `disable_hash` stuff is a behaviour change that probably warrants discussion

Reply:
I like your idea about how to handle error messages. If we had that, I don't think we should change this though - better to ship no error message than a small unreachable one